app/models/katello/glue/pulp/repos.rb

```diff
@@ -252,8 +252,9 @@ def update_repositories
         end
       end
 
-      def add_repo(label, name, url, repo_type, unprotected = false, gpg = nil, checksum_type = nil)
+      def add_repo(label, name, url, repo_type, unprotected = false, gpg = nil, checksum_type = nil, download_policy = nil)
         unprotected = unprotected.nil? ? false : unprotected
+        download_policy = Setting[:default_download_policy] if download_policy.blank? && yum?
```

Comment:
`self` in this context is a product not a repository thus `yum?` fails here with no method error. I think you'll need to check the `repo_type` parameter passed in.